sklearn/utils/validation.py

```diff
@@ -498,10 +498,10 @@ def check_X_y(X, y, accept_sparse=None, dtype="numeric", order=None, copy=False,
                     ensure_2d, allow_nd, ensure_min_samples,
                     ensure_min_features, warn_on_dtype, estimator)
     if multi_output:
-        y = check_array(y, 'csr', force_all_finite=True, ensure_2d=False,
-                        dtype=None)
+        y = check_array(y, 'csr', copy=copy, force_all_finite=True,
+                        ensure_2d=False, dtype=None)
```

Comment:
> Yes, I suppose we should pass these arguments as well.

No, the dtype should not be passed. For classification problems it is very often the case that only X should be converted to a float dtype while y should stay integer or an array of labels (strings using the object dtype).


+1 for documenting which option impacts which array more explicitly though.